src/Makefile

```diff
@@ -115,12 +115,22 @@ else
 ifeq ($(SANITIZER),thread)
 	CFLAGS+=-fsanitize=thread -fno-sanitize-recover=all -fno-omit-frame-pointer
 	LDFLAGS+=-fsanitize=thread
+else
+ifeq ($(SANITIZER),memory)
+ifeq (clang, $(CLANG))
+	MALLOC=libc # MSan provides its own allocator so make sure not to use jemalloc as they clash
+	CFLAGS+=-fsanitize=memory -fsanitize-memory-track-origins=2 -fno-sanitize-recover=all -fno-omit-frame-pointer
+	LDFLAGS+=-fsanitize=memory
+else
+    $(error "MemorySanitizer needs to be compiled and linked with clang. Use CC=clang CXX=clang LD=clang.")
+endif
```

Comment:
I wonder if we can make it even more convenient. If we see the compiler is clang, can we just do:
```
CC=$(CLANG)
CXX=$(CLANG)
LD=$(CLANG)
```
so we just call `make SANITIZER=memory`. Though, these updates won't be propagated to dependecies right? It wouldn't work.

Reply:
What if the user does not have clang installed? Then he we'll get an error that `clang is not a program` or something like that. I think this error is more clear.
Also, yes - I'm not sure if the update will go to deps, but I haven't tested.

Reply:
```
ifeq (clang, $(CLANG))
        CC=$(CLANG) 
        CXX=$(CLANG) 
        LD=$(CLANG)
	MALLOC=libc # MSan provides its own allocator so make sure not to use jemalloc as they clash
	CFLAGS+=-fsanitize=memory -fsanitize-memory-track-origins=2 -fno-sanitize-recover=all -fno-omit-frame-pointer
	LDFLAGS+=-fsanitize=memory
else
    $(error "MemorySanitizer needs to be compiled and linked with clang. Use CC=clang CXX=clang LD=clang.")
endif
```

I meant something like this. It will still check existence of clang, if it exists, we can set CC, CXX, LD